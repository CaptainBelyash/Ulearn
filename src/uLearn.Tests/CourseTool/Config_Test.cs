using System.IO;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using Ulearn.Common.Extensions;
using uLearn.CSharp;

namespace uLearn.CourseTool
{
	[TestFixture]
	public class Config_Test
	{
		[OneTimeSetUp]
		public void SetUp()
		{
			Approvals.RegisterDefaultNamerCreation(() => new RelativeUnitTestFrameworkNamer());
		}

		[Test]
		[UseReporter(typeof(DiffReporter))]
		public void ConfigTemplate_IsOk()
		{
			var path = Path.Combine(TestsHelper.TestDirectory, "templates", "config.xml");
			var config = new FileInfo(path).DeserializeXml<Config>();
			config.IgnoredUlearnSlides = new[] { "1", "2" };
			Approvals.VerifyXml(config.XmlSerialize());
		}
	}
}