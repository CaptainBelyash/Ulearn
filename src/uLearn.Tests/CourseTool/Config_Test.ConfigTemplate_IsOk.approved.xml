<Config>
  <Organization>UrFU</Organization>
  <LtiId>edx</LtiId>
  <ULearnCourseId>BasicProgramming</ULearnCourseId>
  <Video>video.json</Video>
  <Profiles>
    <Profile name="default">
      <UlearnBaseUrlWeb>https://ulearn.me</UlearnBaseUrlWeb>
      <UlearnBaseUrlApi>https://api.ulearn.me</UlearnBaseUrlApi>
    </Profile>
    <Profile name="dev">
      <UlearnBaseUrlWeb>http://localhost:3000</UlearnBaseUrlWeb>
      <UlearnBaseUrlApi>http://localhost:8000</UlearnBaseUrlApi>
    </Profile>
  </Profiles>
  <IgnoredUlearnSlides>
    <SlideId>1</SlideId>
    <SlideId>2</SlideId>
  </IgnoredUlearnSlides>
  <EmitSequentialsForInstructorNotes>false</EmitSequentialsForInstructorNotes>
</Config>